### ISSUE TITLE
Turn WF-API into gradle project

### ISSUE THREAD

**Issue by blankdots** [ATTX-project/workflow-component#25]
#### Description
Use pygradle to handle dependencies and packaging and copying artifacts to the wf-API.
#### DoD
wf-API is built, test and run from gradle.
#### Testing
Peer review.

**Comment by blankdots** [comment]
Requires the pivy repository locally check build.gradle
* download from https://bintray.com/linkedin/maven/pivy-importer#files/com/linkedin/pygradle/pivy-importer/0.3.37
* run as below
```
java -jar ~/Software/pivy-importer-0.3.37-all.jar --repo /home/stefanne/Software/pivy virtualenv:15.0.1 pip:7.1.2 connexion:1.0.129 gunicorn:19.6.0 lxml:3.6.4 rdflib:4.2.1 rdflib-jsonld:0.4.0 PyMySQL:0.7.9 setuptools:19.1.1 swagger-spec-validator:2.0.2 pathlib:1.0.1 wheel:0.26.0 setuptools-git:1.1 flake8:2.5.4 Sphinx:1.4.1 pex:1.1.4 pytest:2.9.1 pytest-cov:2.2.1 pytest-xdist:1.14 setuptools:28.0.0 --replace alabaster:0.7=alabaster:0.7.1,pytz:0a=pytz:2016.4,Babel:0.8=Babel:1.0,sphinx_rtd_theme:0.1=sphinx_rtd_theme:0.1.1,idna:2.0.0=idna:2.1,chardet:2.2=chardet:2.3,setuptools:0.6a2=setuptools:32.1.0

```
Build works following the instructions:
* `gradle wrapper --gradle-version 3.0 `
* `./gradlew build`
* for the tests to run they require a working db server.


```
./gradlew build
:pinRequirements
:createVirtualEnvironment
Already using interpreter /usr/bin/python
New python executable in /home/stefanne/dev/attx/workflow-component/wf-API/wfapi-server/build/venv/bin/python
Installing setuptools, pip, wheel...done.
:installLinks
:installSetupRequirements
Install setuptools-19.1.1 ........................................... [STARTING]
Install (0:01.447 s) ................................................ [FINISHED]
Install wheel-0.26.0 ................................................ [STARTING]
Install (0:00.896 s) ................................................ [FINISHED]
Install pip-7.1.2 ................................................... [STARTING]
Install (0:02.208 s) ................................................ [FINISHED]
Install setuptools-git-1.1 .......................................... [STARTING]
Install (0:00.851 s) ................................................ [FINISHED]
Install pbr-1.8.0 ................................................... [STARTING]
Install (0:00.861 s) ................................................ [FINISHED]
Install argparse-1.4.0 .............................................. [STARTING]
Install (0:00.834 s) ................................................ [FINISHED]
:installBuildRequirements
Install Babel-1.3 ................................................... [STARTING]
Install (0:02.423 s) ................................................ [FINISHED]
Install Jinja2-2.3 .................................................. [STARTING]
Install (0:00.885 s) ................................................ [FINISHED]
Install Pygments-2.0 ................................................ [STARTING]
Install (0:01.553 s) ................................................ [FINISHED]
Install Sphinx-1.4.1 ................................................ [STARTING]
Install (0:02.116 s) ................................................ [FINISHED]
Install alabaster-0.7.1 ............................................. [STARTING]
Install (0:00.841 s) ................................................ [FINISHED]
Install colorama-0.3.5 .............................................. [STARTING]
Install (0:00.810 s) ................................................ [FINISHED]
Install docutils-0.11 ............................................... [STARTING]
Install (0:01.542 s) ................................................ [FINISHED]
Install flake8-2.5.4 ................................................ [STARTING]
Install (0:00.798 s) ................................................ [FINISHED]
Install imagesize-0.7.1 ............................................. [STARTING]
Install (0:00.703 s) ................................................ [FINISHED]
Install mccabe-0.2.1 ................................................ [STARTING]
Install (0:00.814 s) ................................................ [FINISHED]
Install pep8-1.5.7 .................................................. [STARTING]
Install (0:00.742 s) ................................................ [FINISHED]
Install pex-1.1.4 ................................................... [STARTING]
Install (0:00.867 s) ................................................ [FINISHED]
Install pyflakes-0.8.1 .............................................. [STARTING]
Install (0:00.798 s) ................................................ [FINISHED]
Install pytz-2016.4 ................................................. [STARTING]
Install (0:01.616 s) ................................................ [FINISHED]
Install setuptools-19.1.1 ........................................... [SKIPPING]
Install six-1.10.0 .................................................. [STARTING]
Install (0:00.730 s) ................................................ [FINISHED]
Install snowballstemmer-1.1.0 ....................................... [STARTING]
Install (0:00.799 s) ................................................ [FINISHED]
:installPythonRequirements
Install pathlib-1.0.1 ............................................... [STARTING]
Install (0:00.768 s) ................................................ [FINISHED]
Install connexion-1.0.129 ........................................... [STARTING]
Install (0:01.083 s) ................................................ [FINISHED]
Install beautifulsoup4-4.5.1 ........................................ [STARTING]
Install (0:00.800 s) ................................................ [FINISHED]
Install gunicorn-19.6.0 ............................................. [STARTING]
Install (0:00.984 s) ................................................ [FINISHED]
Install lxml-3.6.4 .................................................. [STARTING]
Install (0:57.356 s) ................................................ [FINISHED]
Install rdflib-4.2.1 ................................................ [STARTING]
Install (0:02.623 s) ................................................ [FINISHED]
Install rdflib-jsonld-0.4.0 ......................................... [STARTING]
Install (0:00.827 s) ................................................ [FINISHED]
Install PyMySQL-0.7.9 ............................................... [STARTING]
Install (0:00.884 s) ................................................ [FINISHED]
Install jsonschema-2.5.1 ............................................ [STARTING]
Install (0:01.113 s) ................................................ [FINISHED]
Install Flask-0.10.1 ................................................ [STARTING]
Install (0:01.009 s) ................................................ [FINISHED]
Install PyYAML-3.11 ................................................. [STARTING]
Install (0:01.148 s) ................................................ [FINISHED]
Install requests-2.9.1 .............................................. [STARTING]
Install (0:01.094 s) ................................................ [FINISHED]
Install six-1.10.0 .................................................. [SKIPPING]
Install strict-rfc3339-0.6 .......................................... [STARTING]
Install (0:00.760 s) ................................................ [FINISHED]
Install swagger-spec-validator-2.0.2 ................................ [STARTING]
Install (0:00.819 s) ................................................ [FINISHED]
Install clickclick-1.2 .............................................. [STARTING]
Install (0:00.704 s) ................................................ [FINISHED]
Install isodate-0.5.4 ............................................... [STARTING]
Install (0:00.864 s) ................................................ [FINISHED]
Install pyparsing-2.1.10 ............................................ [STARTING]
Install (0:01.064 s) ................................................ [FINISHED]
Install SPARQLWrapper-1.8.0 ......................................... [STARTING]
Install (0:00.852 s) ................................................ [FINISHED]
Install html5lib-1.0b10 ............................................. [STARTING]
Install (0:01.051 s) ................................................ [FINISHED]
Install argparse-1.4.0 .............................................. [SKIPPING]
Install repoze.lru-0.6 .............................................. [STARTING]
Install (0:00.776 s) ................................................ [FINISHED]
Install functools32-3.2.3-2 ......................................... [STARTING]
Install (0:00.757 s) ................................................ [FINISHED]
Install Werkzeug-0.7 ................................................ [STARTING]
Install (0:01.096 s) ................................................ [FINISHED]
Install Jinja2-2.4 .................................................. [STARTING]
Install (0:01.077 s) ................................................ [FINISHED]
Install itsdangerous-0.21 ........................................... [STARTING]
Install (0:00.805 s) ................................................ [FINISHED]
Install setuptools-19.1.1 ........................................... [SKIPPING]
Install click-6.6 ................................................... [STARTING]
Install (0:00.841 s) ................................................ [FINISHED]
Install webencodings-0.5 ............................................ [STARTING]
Install (0:00.766 s) ................................................ [FINISHED]
Install ordereddict-1.1 ............................................. [STARTING]
Install (0:00.761 s) ................................................ [FINISHED]
:installTestRequirements
Install Flask-0.10.1 ................................................ [SKIPPING]
Install Jinja2-2.4 .................................................. [SKIPPING]
Install PyMySQL-0.7.9 ............................................... [SKIPPING]
Install PyYAML-3.11 ................................................. [SKIPPING]
Install SPARQLWrapper-1.8.0 ......................................... [SKIPPING]
Install Werkzeug-0.7 ................................................ [SKIPPING]
Install argparse-1.4.0 .............................................. [SKIPPING]
Install beautifulsoup4-4.5.1 ........................................ [SKIPPING]
Install click-6.6 ................................................... [SKIPPING]
Install clickclick-1.2 .............................................. [SKIPPING]
Install colorama-0.3.7 .............................................. [STARTING]
Install (0:00.831 s) ................................................ [FINISHED]
Install connexion-1.0.129 ........................................... [SKIPPING]
Install coverage-3.7.1 .............................................. [STARTING]
Install (0:01.067 s) ................................................ [FINISHED]
Install functools32-3.2.3-2 ......................................... [STARTING]
Install (0:00.777 s) ................................................ [FINISHED]
Install gunicorn-19.6.0 ............................................. [SKIPPING]
Install html5lib-1.0b10 ............................................. [SKIPPING]
Install isodate-0.5.4 ............................................... [SKIPPING]
Install itsdangerous-0.21 ........................................... [SKIPPING]
Install jsonschema-2.5.1 ............................................ [SKIPPING]
Install lxml-3.6.4 .................................................. [SKIPPING]
Install nose-1.3.7 .................................................. [STARTING]
Install (0:01.151 s) ................................................ [FINISHED]
Install ordereddict-1.1 ............................................. [SKIPPING]
Install pathlib-1.0.1 ............................................... [SKIPPING]
Install py-1.4.29 ................................................... [STARTING]
Install (0:00.922 s) ................................................ [FINISHED]
Install pyparsing-2.1.10 ............................................ [SKIPPING]
Install pytest-cov-2.2.1 ............................................ [STARTING]
Install (0:00.818 s) ................................................ [FINISHED]
Install pytest-xdist-1.14 ........................................... [STARTING]
Install (0:01.487 s) ................................................ [FINISHED]
Install pytest-2.9.1 ................................................ [STARTING]
Install (0:01.229 s) ................................................ [FINISHED]
Install rdflib-jsonld-0.4.0 ......................................... [SKIPPING]
Install rdflib-4.2.1 ................................................ [SKIPPING]
Install repoze.lru-0.6 .............................................. [SKIPPING]
Install requests-2.9.1 .............................................. [SKIPPING]
Install setuptools-19.1.1 ........................................... [SKIPPING]
Install six-1.10.0 .................................................. [SKIPPING]
Install strict-rfc3339-0.6 .......................................... [SKIPPING]
Install swagger-spec-validator-2.0.2 ................................ [SKIPPING]
Install webencodings-0.5 ............................................ [SKIPPING]
:installProject
Install wfapi-server ................................................ [STARTING]
Install (0:04.064 s) ................................................ [FINISHED]
:buildWheels
Preparing wheel Flask-0.10.1 ........................................ [STARTING]
Preparing wheel Flask-0.10.1 (0:01.045 s) ........................... [FINISHED]
Preparing wheel Jinja2-2.4 .......................................... [STARTING]
Preparing wheel Jinja2-2.4 (0:00.994 s) ............................. [FINISHED]
Preparing wheel PyMySQL-0.7.9 ....................................... [STARTING]
Preparing wheel PyMySQL-0.7.9 (0:00.846 s) .......................... [FINISHED]
Preparing wheel PyYAML-3.11 ......................................... [STARTING]
Preparing wheel PyYAML-3.11 (0:01.192 s) ............................ [FINISHED]
Preparing wheel SPARQLWrapper-1.8.0 ................................. [STARTING]
Preparing wheel SPARQLWrapper-1.8.0 (0:00.800 s) .................... [FINISHED]
Preparing wheel Werkzeug-0.7 ........................................ [STARTING]
Preparing wheel Werkzeug-0.7 (0:01.111 s) ........................... [FINISHED]
Preparing wheel argparse-1.4.0 ...................................... [STARTING]
Preparing wheel argparse-1.4.0 (0:00.742 s) ......................... [FINISHED]
Preparing wheel beautifulsoup4-4.5.1 ................................ [STARTING]
Preparing wheel beautifulsoup4-4.5.1 (0:00.894 s) ................... [FINISHED]
Preparing wheel click-6.6 ........................................... [STARTING]
Preparing wheel click-6.6 (0:00.983 s) .............................. [FINISHED]
Preparing wheel clickclick-1.2 ...................................... [STARTING]
Preparing wheel clickclick-1.2 (0:00.720 s) ......................... [FINISHED]
Preparing wheel connexion-1.0.129 ................................... [STARTING]
Preparing wheel connexion-1.0.129 (0:01.030 s) ...................... [FINISHED]
Preparing wheel functools32-3.2.3-2 ................................. [STARTING]
Preparing wheel functools32-3.2.3-2 (0:00.750 s) .................... [FINISHED]
Preparing wheel gunicorn-19.6.0 ..................................... [STARTING]
Preparing wheel gunicorn-19.6.0 (0:01.092 s) ........................ [FINISHED]
Preparing wheel html5lib-1.0b10 ..................................... [STARTING]
Preparing wheel html5lib-1.0b10 (0:00.931 s) ........................ [FINISHED]
Preparing wheel isodate-0.5.4 ....................................... [STARTING]
Preparing wheel isodate-0.5.4 (0:00.809 s) .......................... [FINISHED]
Preparing wheel itsdangerous-0.21 ................................... [STARTING]
Preparing wheel itsdangerous-0.21 (0:00.743 s) ...................... [FINISHED]
Preparing wheel jsonschema-2.5.1 .................................... [STARTING]
Preparing wheel jsonschema-2.5.1 (0:01.204 s) ....................... [FINISHED]
Preparing wheel lxml-3.6.4 .......................................... [STARTING]
Preparing wheel lxml-3.6.4 (1:00.538 s) ............................. [FINISHED]
Preparing wheel ordereddict-1.1 ..................................... [STARTING]
Preparing wheel ordereddict-1.1 (0:00.779 s) ........................ [FINISHED]
Preparing wheel pathlib-1.0.1 ....................................... [STARTING]
Preparing wheel pathlib-1.0.1 (0:00.880 s) .......................... [FINISHED]
Preparing wheel pyparsing-2.1.10 .................................... [STARTING]
Preparing wheel pyparsing-2.1.10 (0:01.114 s) ....................... [FINISHED]
Preparing wheel rdflib-jsonld-0.4.0 ................................. [STARTING]
Preparing wheel rdflib-jsonld-0.4.0 (0:00.850 s) .................... [FINISHED]
Preparing wheel rdflib-4.2.1 ........................................ [STARTING]
Preparing wheel rdflib-4.2.1 (0:02.553 s) ........................... [FINISHED]
Preparing wheel repoze.lru-0.6 ...................................... [STARTING]
Preparing wheel repoze.lru-0.6 (0:00.780 s) ......................... [FINISHED]
Preparing wheel requests-2.9.1 ...................................... [STARTING]
Preparing wheel requests-2.9.1 (0:00.911 s) ......................... [FINISHED]
Preparing wheel setuptools-19.1.1 ................................... [STARTING]
Preparing wheel setuptools-19.1.1 (0:00.916 s) ...................... [FINISHED]
Preparing wheel six-1.10.0 .......................................... [STARTING]
Preparing wheel six-1.10.0 (0:01.003 s) ............................. [FINISHED]
Preparing wheel strict-rfc3339-0.6 .................................. [STARTING]
Preparing wheel strict-rfc3339-0.6 (0:00.804 s) ..................... [FINISHED]
Preparing wheel swagger-spec-validator-2.0.2 ........................ [STARTING]
Preparing wheel swagger-spec-validator-2.0.2 (0:00.734 s) ........... [FINISHED]
Preparing wheel webencodings-0.5 .................................... [STARTING]
Preparing wheel webencodings-0.5 (0:00.761 s) ....................... [FINISHED]
:generateCompletions
:buildPex
Processing /home/stefanne/dev/attx/workflow-component/wf-API/wfapi-server
Building wheels for collected packages: wfapi-server
  Running setup.py bdist_wheel for wfapi-server
  Stored in directory: /home/stefanne/dev/attx/workflow-component/wf-API/wfapi-server/build/wheel-cache
Successfully built wfapi-server
/home/stefanne/dev/attx/workflow-component/wf-API/wfapi-server/build/deployable/bin/wfapi-server.pex
Processing entry point: wfapi = wf_api.wfapi:main
:packageDeployable
:assemble
:flake8
:pytest
============================= test session starts ==============================
platform linux2 -- Python 2.7.6, pytest-2.9.1, py-1.4.29, pluggy-0.3.1
rootdir: /home/stefanne/dev/attx/workflow-component/wf-API/wfapi-server, inifile: 
plugins: cov-2.2.1
collected 17 items

tests/test_activity.py ......
tests/test_activity_metadata.py .
tests/test_app.py .
tests/test_parse_config.py ..
tests/test_workflow.py ......
tests/test_workflow_metadata.py .

========================== 17 passed in 0.33 seconds ===========================
:check
:build

BUILD SUCCESSFUL

Total time: 3 mins 41.606 secs
```

**Comment by blankdots** [comment]
for Python3 it fails at package `functools32-3.2.3-2` (this should only be available for python 2.7) most likely because `connexion` module.

**Comment by blankdots** [comment]
One part missing is the html reports generation, it requires a gradle task.
